docs/user-guide/src/ironic/ironic-python-agent.md

```diff
@@ -0,0 +1,33 @@
+# ironic python agent (IPA)
+
+IPA is a service written in python that runs within a ramdisk. It provides remote access to  `ironic-conductor` and `ironic-inspector` services to perform various operations on the managed server. It also sends information about the server to `Ironic`.
+
+
+## Data flow
+
+IPA interacts with other components. The information exchanged and the component to which it is sent to or received from are described below.
+The communication between IPA and these component can be encrypted in-transit with SSL/TLS.
+
+- Heartbeat: periodic message informing Ironic that the node is still running.
+- Lookup: data sent to Ironic that helps it determine Ironic’s node UUID for the node.
+- Introspection result: data about hardware details, such as CPU, disk, RAM and Network interfaces.
```

Comment:
s/ result// (for consistency with first 2 points). s/Network/network/, s/data about//